frontend/src/app/choreography-frame.ts

```diff
@@ -2,4 +2,5 @@ import { ChoreographySubframe } from './choreography-subframe';
 
 export interface ChoreographyFrame {
   subframes: ChoreographySubframe[];
+  notes?: string;
```

Comment:
**issue:** optional fields are evil (you need to always check for their existence etc.). Make everything mandatory where possible and initialize as empty where a sane empty value is available (`''` in this case)